osu.Game/Overlays/Chat/ChannelListItem.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Overlays.Chat
+{
+    public class ChannelListItem : ClickableContainer, IFilterable
+    {
+        private const float width_padding = 5;
+        private const float channel_width = 150;
+        private const float text_size = 15;
+        private const float transition_duration = 100;
+
+        private readonly OsuSpriteText topic;
+        private readonly OsuSpriteText name;
+        private readonly TextAwesome joinedCheckmark;
+
+        private Color4? joinedColour;
+        private Color4? topicColour;
+
+        public string[] FilterTerms => new[] { Channel.Name };
+        public bool MatchingCurrentFilter
+        {
+            set
+            {
+                FadeTo(value ? 1f : 0f, 100);
+            }
+        }
+
+        public Action<Channel> OnRequestJoin;
+        public Action<Channel> OnRequestLeave;
+
+        private Channel channel;
+        public Channel Channel

```

Comment:
just remove this property. i can't think of a case we'd want to change the channel being displayed by a `ChannelListItem`